test/built-ins/TypedArray/prototype/sort/sortcompare-with-no-tostring.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.sort
+description: TypedArrays sort does not cast values to String
+info: >
+  22.2.3.26 %TypedArray%.prototype.sort ( comparefn )
+
+  When the TypedArray SortCompare abstract operation is called with two
+  arguments x and y, the following steps are taken:
+
+  ...
+  2. If the argument comparefn is not undefined, then
+    a. Let v be ? Call(comparefn, undefined, « x, y »).
+    ...
+  ...
+includes: [testTypedArray.js, compareArray.js]
+---*/
+
+var origToString = Number.prototype.toString;
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([4, 3, 2, 1]);
+
+  Number.prototype.toString = function() {
+    throw new Test262Error();
+  };
```

Comment:
This is a good idea, but I think it is unlikely to catch the kind of bugs
  that you have in mind. The ToString internal operation does not invoke
  Number.prototype.toString, so even if SortCompare for TypedArrays incorrectly
  used the semantics for Arrays, this test would pass.

  One way to ensure that ToString is not being called on the elements is to
  select values that would sort differently when interpreted as their string
representations, e.g.

  ```js
  var sample = new TA([20, 100, 3]);
  ```